docs/toolchain/index.rst

```diff
@@ -0,0 +1,55 @@
+.. _sec-toolchain:
+
+Toolchain
+=========
+
+The toolchain (compiler, archiver, and linker) used for building **xNVMe**
+must support **C11**, **pthreads** and on the system the following tools must
+be available:
+
+* Python (>=3.7)
+* meson (>=0.58) and matching version of ninja
+* make (gmake)
+* gcc/mingw/clang
+
+Along with libraries:
+
+* glibc (>= 2.28, for **io_uring/liburing**)
+* libaio-dev (>=0.3, For **xNVMe** and **SPDK**)
+* libnuma-dev (>=2, For **SPDK**)
+* libssl-dev (>=1.1, For **SPDK**)
+* liburing (>=2.2, for **xNVMe**)
+* uuid-dev (>=2.3, For **SPDK**)
```

Comment:
Right, this is actually very frustrating as there are so many finicky things to explain her, e.g. that glibc version is there to satisfy liburing, that liburing is needed for SPDK and xNVMe, that libssl-dev is needed by SPDK not by xNVMe etc etc. I think I will wipe this from the docs and instead explain the toolchain section as a build **xNVMe** with all bells-and-whistles, and leave it to  all the toolchain scripts to provide the information as this is updated since it is verified by CI.

Regarding this, then I realized a behavior change of the build system would be preferable, described here:
https://github.com/xnvme/xnvme/issues/508

Reply:
I agree with this approach